packages/cardpay-sdk/providers/json-rpc-provider.ts

```diff
@@ -5,13 +5,17 @@ import { Networkish } from '@ethersproject/networks';
 
 const errorGas = ['call', 'estimateGas'];
 
+function isExpectedMessage(message: string) {
+  return message.toLocaleLowerCase().match('reverted') || message.match(/0x.+/);
+}
+
 function spelunk(value: any): null | { message: string; data: string } {
   if (value == null) {
     return null;
   }
 
   // These *are* the droids we're looking for.
-  if (typeof value.message === 'string' && value.message.match('reverted') && isHexString(value.data)) {
+  if (typeof value.message === 'string' && (isExpectedMessage(value.message) || isHexString(value.data))) {
```

Comment:
isn't the problem that you also need to check `value.data.data`--from your description, it sounded like the revert data was nested in an extra `data` object?

Reply:
or wait never mind--i see that spelunk is recursive, so it will eventually reach the right place.